File: src/main/kotlin/by/zmitser/webapp/taskapp/domains/Task.kt
package by.zmitser.webapp.taskapp.domains

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDate

@Document
data class Task(@Id var id: String?, var name: String?, var dueDate: LocalDate?, var completed: Boolean?) {

    constructor() : this(null, null, null, null)
    constructor(name: String?, dueDate: LocalDate?, completed: Boolean?) : this(null, name, dueDate, completed)
}

File: src/main/front/src/app/tasks/task.ts
export class Task {
    public id: string;
    public name: string;
    public completed: boolean;
    public dueDate: string;

    constructor(id: string, name: string, completed: boolean, dueDate: string) {
        this.id = id;
        this.name = name;
        this.completed = completed;
        this.dueDate = dueDate;
    }

}
